doc/source/building/macosx.rst

```diff
@@ -48,12 +48,37 @@ BLAS/LAPACK Installation
 
 You will also need to install a library providing the BLAS and LAPACK
 interfaces. ATLAS, OpenBLAS, and MKL all work. OpenBLAS can be installed
-via `Homebrew <https://brew.sh/>`.
+via `Homebrew <https://brew.sh/>`_.
 
 As of SciPy version 1.2.0, we do not support compiling against the system
 Accelerate library for BLAS and LAPACK. It does not support a sufficiently
 recent LAPACK interface.
 
+Install libraries from Homebrew:
+
+.. code-block:: shell
+
+     brew install openblas lapack
+
+Before installing SciPy, you might also need to setup environment variables
+that tell where to LAPACK library, and also some C compiler options
+to make it pass on macOS clang:
+
+.. code-block:: shell
+
+    # Was needed on macOS Big Sur 11.6
+    export CFLAGS=-Wno-error=implicit-function-declaration
+
+    # Tell where to find LAPACK include files for building C interfaces
+    export LDFLAGS="-L/usr/local/opt/lapack/lib"
+    export CPPFLAGS="-I/usr/local/opt/lapack/include"
+    export PKG_CONFIG_PATH="/usr/local/opt/lapack/lib/pkgconfig"
+
+    export LAPACK=/usr/local/opt/lapack/lib/liblapack.dylib
+    export BLAS=/usr/local/opt/openblas/lib/libblas.dylib 
+
+`More information in this StackOverflow answer <https://stackoverflow.com/a/70880741/315168>`_.
+
```

Comment:
Improved docs are always welcome. One thing to keep in mind is that we're in the middle of a transition to a new build system called `meson`, so folks may not prioritize improving the docs for the old one too much.

One other thing to note is that we have a file `site.cfg.example` which can be edited to `site.cfg` to point to some of the paths you mention above to select the linear algebra backend that gets used.  Perhaps even more convenient is that you can use `~/.numpy-site.cfg` to have a user-fixed OpenBLAS backend specified for i.e., both NumPy and SciPy.

Mine looks like this for example (` ~/.numpy-site.cfg`):

```
[openblas]
libraries = openblas
library_dirs = /Users/treddy/github_projects/spack/opt/spack/darwin-catalina-skylake/gcc-8.3.0/openblas-0.3.9-sasoit4i4h37ohcdq7rkmani4shqmhqf/lib
include_dirs = /Users/treddy/github_projects/spack/opt/spack/darwin-catalina-skylake/gcc-8.3.0/openblas-0.3.9-sasoit4i4h37ohcdq7rkmani4shqmhqf/include
runtime_library_dirs = /Users/treddy/github_projects/spack/opt/spack/darwin-catalina-skylake/gcc-8.3.0/openblas-0.3.9-sasoit4i4h37ohcdq7rkmani4shqmhqf/lib
```